### ISSUE TITLE
Look at COLORFGBG for automatic background again (improved)

### ISSUE THREAD

**Issue by blueyed** [neovim/neovim#2764]
Commit 3baba1e removed the feature where Vim looks at `$COLORFGBG` to auto-set the background to light/dark.

```
/*
 * Return "dark" or "light" depending on the kind of terminal.
 * This is just guessing!  Recognized are:
 * "linux"         Linux console
 * "screen.linux"   Linux console with screen
 * "cygwin"        Cygwin shell
 * "putty"         Putty program
 * We also check the COLORFGBG environment variable, which is set by
 * rxvt and derivatives. This variable contains either two or three
 * values separated by semicolons; we want the last value in either
 * case. If this value is 0-6 or 8, our background is dark.
 */
static char_u *term_bg_default(void)
{
  char_u      *p;

  if (STRCMP(T_NAME, "linux") == 0
      || STRCMP(T_NAME, "screen.linux") == 0
      || STRCMP(T_NAME, "cygwin") == 0
      || STRCMP(T_NAME, "putty") == 0
      || ((p = (char_u *)os_getenv("COLORFGBG")) != NULL
          && (p = vim_strrchr(p, ';')) != NULL
          && ((p[1] >= '0' && p[1] <= '6') || p[1] == '8')
          && p[2] == NUL))
    return (char_u *)"dark";
  return (char_u *)"light";
}
```

It would be nice to re-add this, but improve it (if possible), so that the
real value of a color would be used, i.e. depending on the terminal palette.

**Comment by blueyed** [comment]
This should probably also try to query the terminals background color, e.g. for urxvt:

    printf '\e]11;?\e\\' | cat -A

**Comment by egmontkob** [comment]
See also https://bugzilla.gnome.org/show_bug.cgi?id=733423

**Comment by dquach217** [comment]
@justinmk can I work on this?

**Comment by justinmk** [comment]
@dquach217 certainly! In the future there's no need to ask, just post a PR (search existing PRs to avoid duplicate work/discussions) :)

**Comment by justinmk** [comment]
Looks like urxvt and gnome-terminal support the `"\x1b]11;?\x07"` request, so there's no need to implement `$COLORFGBG` scraping.

Closed by #9509
